core/bin/external_node/src/config/observability.rs

```diff
@@ -1,43 +1,99 @@
-use zksync_config::configs::ObservabilityConfig;
-
-pub fn observability_config_from_env() -> anyhow::Result<ObservabilityConfig> {
-    // The logic in this method mimics the historical logic of loading observability options
-    // This is left intact, since some of the existing deployments may rely on the this behavior.
-    let sentry_url = if let Ok(sentry_url) = std::env::var("MISC_SENTRY_URL") {
-        if sentry_url == "unset" {
-            None
-        } else {
-            Some(sentry_url)
+use std::{collections::HashMap, time::Duration};
+
+use anyhow::Context as _;
+use prometheus_exporter::PrometheusExporterConfig;
+use serde::Deserialize;
+use vlog::LogFormat;
+
+use super::{ConfigurationSource, Environment};
+
+/// Observability part of the node configuration.
+#[derive(Debug, Default, Deserialize)]
+pub(crate) struct ObservabilityENConfig {
+    /// Port to bind the Prometheus exporter server to. If not specified, the server will not be launched.
+    /// If the push gateway URL is specified, it will prevail.
+    pub prometheus_port: Option<u16>,
+    /// Prometheus push gateway to push metrics to. Overrides `prometheus_port`. A full URL must be specified
+    /// including `job_id` and other path segments; it will be used verbatim as the URL to push data to.
+    pub prometheus_pushgateway_url: Option<String>,
```

Comment:
Is this approach reasonable? I don't really like how the gateway URL is opaquely constructed in `PrometheusConfig`.